lib/js/src/manager/screen/utils/AudioData.js

```diff
@@ -0,0 +1,159 @@
+                || phoneticType === SpeechCapabilities.SC_TEXT || phoneticType === SpeechCapabilities.PRE_RECORDED)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Adds audio files to _audioFiles.
+     * @param {SdlFile[]} audioFiles - A list of audio files referenced by SdlFile instances.
+     */
+    addAudioFiles (audioFiles) {
+        if (this._audioFiles === null || this._audioFiles === undefined) {
+            this._audioFiles = new Map();
+        }
+        audioFiles.map((audioFile) => {
+            this._audioData.push(new TTSChunk().setText(audioFile.getName()).setType(SpeechCapabilities.FILE));
+            this._audioFiles.set(audioFile.getName(), audioFile);
+        });
+    }
+
+    /**
+     * Create additional strings to be spoken by the system speech synthesizer.
+     * @param {String[]} spokenString - The strings to be spoken by the system speech synthesizer.
+     */
+    addSpeechSynthesizerStrings (spokenString) {
+        if (spokenString.length === 0) {
+            return;
+        }
+        const newPrompts = [];
+        for (const spoken of spokenString) {
+            if (spoken.length === 0) {
+                continue;
+            }
+            newPrompts.push(new TTSChunk().setText(spoken));
+        }
+        if (newPrompts.length === 0) {
+            return;
+        }
+        if (this._audioData === null || this._audioData === undefined) {
+            this._audioData = newPrompts;
+            return;
+        }
+        this._audioData.push.apply(this._audioData, newPrompts);
+    }
+
+    /**
+     * Create additional strings to be spoken by the system speech synthesizer using a phonetic string.
+     * @param {String[]} phoneticString - The strings to be spoken by the system speech synthesizer
+     * @param {SpeechCapabilities} phoneticType - Must be one of `SAPI_PHONEMES`, `LHPLUS_PHONEMES`, `TEXT`, or `PRE_RECORDED` or no object will be created
+     */
+    addPhoneticSpeechSynthesizerStrings (phoneticString, phoneticType) {
+        if (!this.isValidPhoneticType(phoneticType) || phoneticString.length === 0) {
+            return;
+        }
+        const newPrompts = [];
+        for (const phonetic of phoneticString) {
+            if (phonetic.length === 0) {
+                continue;
+            }
+            newPrompts.push(new TTSChunk().setText(phonetic).setType(phoneticType));
+        }
+        if (newPrompts.length === 0) {
+            return;
+        }
+        if (this._audioData === null || this._audioData === undefined) {
+            this._audioData = newPrompts;
+            return;
+        }
+        this._audioData.push.apply(this._audioData, newPrompts);
+    }
+
+    /**
+     * Gets the audio files.
+     * @returns {Map} - A HashMap of audio file names and SdlFile instances.
+     */
+    getAudioFiles () {
```

Comment:
```suggestion
    /**
     * Gets the audio files.
     * @private
     * @returns {Map} - A HashMap of audio file names and SdlFile instances.
     */
    _getAudioFiles () {
```
This method is package private in the Java Suite.